components/dropdown-menu/index.js

```diff
@@ -47,10 +50,26 @@ class DropdownMenu extends wp.element.Component {
 		this.closeMenu();
 	}
 
+	handleFocus() {
+		this.hasFocus = true;
+	}
+
+	handleBlur() {
+		this.hasFocus = false;
+
+		this.maybeCloseMenuOnBlur = setTimeout( () => {
+			this.closeMenu();
+		}, 100 );
```

Comment:
Completely arbitrary (I know...). There's a brief interval between blur on an element and focus on the following element. '0' didn't work. On the W3C examples, they use '300', see:
https://www.w3.org/TR/wai-aria-practices/examples/menu-button/js/Menubutton.js
https://www.w3.org/TR/wai-aria-practices/examples/menu-button/js/MenuItemAction.js
https://www.w3.org/TR/wai-aria-practices/examples/menu-button/js/PopupMenuAction.js

Open to suggestions...